File: authorization/src/main/java/com/example/alexeyglushkov/authorization/tools/JsonTokenExtractor.java
package com.example.alexeyglushkov.authorization.tools;

import com.example.alexeyglushkov.authorization.OAuth.Token;

import org.json.JSONException;
import org.json.JSONObject;

public class JsonTokenExtractor implements TokenExtractor
{
  public Token extract(String response)
  {
    Token result = null;
    try {
      JSONObject jsonObject = new JSONObject(response);
      result = new Token(jsonObject.getString("access_token"), "", response);
    } catch (JSONException ex) {

    }

    return result;
  }

}

File: authorization/src/main/java/com/example/alexeyglushkov/authorization/tools/JsonTokenExtractor.kt
package com.example.alexeyglushkov.authorization.tools

import com.example.alexeyglushkov.authorization.OAuth.Token
import org.json.JSONException
import org.json.JSONObject

class JsonTokenExtractor : TokenExtractor {
    override fun extract(response: String?): Token? {
        var result: Token? = null
        try {
            val jsonObject = JSONObject(response)
            result = Token(jsonObject.getString("access_token"), "", response)
        } catch (ex: JSONException) {
        }
        return result
    }
}